jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/pinning/ConnectionPinningManager.kt

```diff
@@ -129,35 +131,37 @@ class DefaultConnectionPinningManager :
 
     @VisibleForTesting
     internal fun showDialogIfNeeded(oldConnection: ToolkitConnection?, newConnection: ToolkitConnection, featuresString: String, project: Project? = null) =
-        if (!doNotPromptForPinning) {
-            val bearerTokenConnectionName = bearerTokenConnectionString(oldConnection, newConnection)
-
-            computeOnEdt(ModalityState.defaultModalityState()) {
-                MessageDialogBuilder.yesNo(
-                    message("credentials.switch.confirmation.title", featuresString, bearerTokenConnectionName),
-                    message("credentials.switch.confirmation.comment", featuresString, bearerTokenConnectionName, message("iam.name"))
-                )
-                    .yesText(message("credentials.switch.confirmation.yes"))
-                    .noText(message("credentials.switch.confirmation.no"))
-                    .doNotAsk(object : com.intellij.openapi.ui.DoNotAskOption.Adapter() {
-                        override fun rememberChoice(isSelected: Boolean, exitCode: Int) {
-                            if (isSelected && exitCode == DialogWrapper.OK_EXIT_CODE) {
-                                doNotPromptForPinning = true
+        shouldPinConnections.let { shouldPinConnections ->
+            if (shouldPinConnections == null) {
+                val bearerTokenConnectionName = bearerTokenConnectionString(oldConnection, newConnection)
```

Comment:
nit: Could we change the function name to getBearerTokenConnectionName? We could do this as part of another PR.